cypress/support/step_definitions/fireboltCalls.js

```diff
@@ -674,3 +674,199 @@ Given(
     }
   }
 );
+
+/**
+ * @module fireboltCalls
+ * @function '(.+)' registers for the '(.*?)'(?: '(.*?)')? event
+ * @description Sending a message to platform or app to register a event
+ * @param {String} appId - app identtifier.
+ * @param {String} sdk - sdk name.
+ * @param {String} fireboltCallKey - key name passed to look for firebolt call object in fireboltCallData.
+ * @example
+ * And '1st party app' registers for the 'Firebolt' 'CLOSEDCAPTION_SETTINGS' event
+ * And '3rd party app' registers for the 'Firebolt' 'CLOSEDCAPTION_SETTINGS' event
+ * And '1st party app' registers for the 'Firebolt' event
+ */
+Given(
+  /'(.+)' registers for the '(.*?)'(?: '(.*?)')? event$/,
+  async (appId, sdk, fireboltCallKey) => {
+    if (CONSTANTS.SUPPORTED_SDK.includes(sdk)) {
+      let fireboltCallObject;
+      let fireboltCallObjectErrorMessage = CONSTANTS.NO_DATA_FOR_THE_KEY + fireboltCallKey;
+
+      // Creating runtime environment variable
+      if (!UTILS.getEnvVariable('runtime', false)) {
+        Cypress.env('runtime', {});
+      }
+
+      // When fireboltCall object key passed fetching the object from the fireboltCalls data else reading it from environment variable
+      // TODO: appending the firebolt object will take care in optimization ticket
+      if (fireboltCallKey) {
+        cy.getFireboltData(fireboltCallKey).then((fireboltData) => {
+          fireboltCallObject = fireboltData;
+          UTILS.getEnvVariable('runtime').fireboltCall = fireboltData;
```

Comment:
Does this work to set like this? I would have thought you'd need to do something like `Cypress.env(field, value)`.

Reply:
Yes, corrected the code.